### PR TITLE
New json flag "CAN_OPEN_DOORS" (fixed)

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

SUMMARY: Features "Implementation of a new json flag 'CAN_OPEN_DOORS'"

#### Purpose of change

Zombie scientists got the flag for now. Cyborgs from #29206 might get the flag later.

#### Describe the solution

Monsters with that flag will try to open a door if there's one blocking their path.
Always tries to open doors before bashing them.

#### Additional context

Remade #28734 because I accidentally deleted the branch for that PR long time ago. Probably that is why it didn't work.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
Suggestion: add this flag to the [prototype cyborg](https://github.com/CleverRaven/Cataclysm-DDA/blob/f078e2d44ab982cf70baa9e534d8185d4be5c633/data/json/monsters.json#L647). For two reasons:
1. It was originally added to this monster, but dropped when the precursor to this PR was not merged
2. It gives you a monster in the base game that you can test that this flag works with

**Comment by Wokko1** [comment]
> #29206 Has already been merged, so perhaps add this flag to the prototype cyborg monster so that you can test the flag and because it was intended in that PR.

Done, prototype cyborgs have the flag now.

File: src/monmove.cpp [hunk 1]
```diff
@@ -725,11 +725,13 @@ void monster::move()
             }
         }
     }
+    bool can_open_doors = has_flag( MF_CAN_OPEN_DOORS );
```
Comment (Night-Pryanik) [inline src/monmove.cpp]:
```suggestion
    const bool can_open_doors = has_flag( MF_CAN_OPEN_DOORS );
```

Comment (Wokko1) [reply src/monmove.cpp]: Thx, done

**Comment by kevingranade** [comment]
How exactly do you get this to work? If I spawn a prototype cyborg and then run through a doorway and close it, I've yet to get one to open the door afterwards.

**Comment by Wokko1** [comment]
> How exactly do you get this to work? If I spawn a prototype cyborg and then run through a doorway and close it, I've yet to get one to open the door afterwards.

It definitely works: https://gfycat.com/CrazyJovialArabianhorse
Though for zombie scientists only. I don't know why, but prototype cyborgs don't open doors. But because z-scientists successfully open doors, I think the bug is not related to my pr? I'm just going to remove the flag from cyborgs then.

**Comment by Fris0uman** [comment]
Could it be that the cyborg doesn't have smell and thus stop following when the player gets out of site ?

**Comment by Wokko1** [comment]
> Could it be that the cyborg doesn't have smell and thus stop following when the player gets out of sight ?

Maybe because of `PATH_AVOID_DANGER_2`.

**Comment by Fris0uman** [comment]
As far as I can tell `PATH_AVOID_DANGER_2` only affect avoiding traps, It should not affect doors.

**Pull request merged**